src/hf_hydrodata/gridded.py

```diff
@@ -1638,17 +1638,20 @@ def _get_ndarray_from_api(entry, options):
 
         try:
             headers = _get_api_headers()
-            response = requests.get(gridded_data_url, headers=headers, timeout=1200)
+            response = requests.get(gridded_data_url, headers=headers, timeout=4000)
+            if response.status_code in [500, 502]:
```

Comment:
502 is a bad gateway error, indicating a problem with the proxy layer. Does that apply here?

Reply:
Never mind, this seems fine. The proxy could time out too I suppose